### PR TITLE
fix(StatusInput): Replaced layout spacing with margins

### PR DESCRIPTION
**Pull request by alexandraB99**
As part of https://github.com/status-im/status-desktop/issues/6022

### Checklist

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [ ] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [x] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [x] test changes in [status-desktop](https://github.com/status-im/status-desktop)

### CODE REVIEW

File: src/StatusQ/Controls/StatusInput.qml [hunk 1]
```diff
@@ -419,7 +424,8 @@ Item {
             height: visible ? contentHeight : 0
             font.pixelSize: 12
             color: Theme.palette.dangerColor1
-            Layout.alignment: Qt.AlignVCenter
+            Layout.alignment: Qt.AlignVCenter | Qt.AlignTop
```
Comment (caybro) [inline src/StatusQ/Controls/StatusInput.qml:427]: `Qt.AlignVCenter | Qt.AlignTop` for real? :) I'm confused, and QML will be as well :D

File: src/StatusQ/Controls/StatusInput.qml [hunk 2]
```diff
@@ -348,7 +348,9 @@ Item {
     }
 
     implicitWidth: 448
-    implicitHeight: (internal.inputHeight + topRow.height + errorMessage.height + (2*inputLayout.spacing))
+    implicitHeight: (internal.inputHeight + topRow.height + errorMessage.height +
+                    ((topRow.height > 0) && (errorMessage.height > 0) ? 16 :
+                    (topRow.height > 0) || (errorMessage.height > 0) ? 8 : 0))
```
Comment (caybro) [inline src/StatusQ/Controls/StatusInput.qml:353]: This is becoming too much... can't we do some `layout.childrenRect` instead to simplify this?

**Review by caybro — CHANGES_REQUESTED** [review]

File: src/StatusQ/Controls/StatusInput.qml (hunk 1 above)
Comment (alexandraB99) [reply src/StatusQ/Controls/StatusInput.qml]: true! vcenter removed

File: src/StatusQ/Controls/StatusInput.qml (hunk 2 above)
Comment (alexandraB99) [reply src/StatusQ/Controls/StatusInput.qml:353]: it didn't work out of the box, at least as a quick 'fix', the layout was not quite reacting to height changes of its children, the 'spaghetti' is just for the spacing to be taken into account.

**Review by caybro — APPROVED** [review]

**Pull request merged**